compiler/parser/RdrHsSyn.hs

```diff
@@ -3028,6 +3028,25 @@ mkTemplateFunBindDecl fname con_name (Just body) binds = do
       match_group = matchGroup loc match
   funBind loc tag match_group
 
+-- | Construct a 'maintainer' function binding. Note for future
+-- refactoring : this only differs from 'mkTemplateFunBindDecl' in the
+-- generated function's argument ('key' vs. 'this').
+mkMaintainerFunBindDecl ::
```

Comment:
Can we have a version parameterised by key vs this, so we only write this once.

Reply:
Yes, but I was in a hurry and so left it as a note for a future refactoring.

Reply:
^ Yesterday was intensely busy and I was under a lot of pressure to turn that change around quickly :(

Reply:
That's fine, no rush to do it now 😄